pkg/project/git.go

```diff
@@ -158,6 +157,12 @@ func findGitConfigFile(fp string, directory string, match string) (string, bool)
 		return "", false
 	}
 
+	recursiveGitCalls++
```

Comment:
Can we replace the recursive call by a while loop (via for in go)? Then we do not need the global variable.

Reply:
In python we had instance variables, but a loop makes more sense in Go.